src/view3d.cpp

```diff
@@ -185,61 +209,155 @@ void View3D::centerOnGeometry(const QModelIndex& index)
     m_camera.setEyeToCenterDistance(std::max<double>(2*m_camera.clipNear(), diag*0.7));
 }
 
-
 void View3D::initializeGL()
 {
+    // GLEW has a problem with core contexts. It calls glGetString(GL_EXTENSIONS)​,
+    // which causes GL_INVALID_ENUM on GL 3.2+ core context as soon as glewInit()​ is called.
+    // It also doesn't fetch the function pointers. The solution is for GLEW to use glGetStringi​ instead.
+    // The current version of GLEW is 1.10.0 but they still haven't corrected it.
+    // The only fix is to use glewExperimental​ for now.
+    // https://www.opengl.org/wiki/OpenGL_Loading_Library
+    glewExperimental = true;
     if (glewInit() != GLEW_OK)
     {
         g_logger.error("%s", "Failed to initialize GLEW");
         m_badOpenGL = true;
         return;
     }
+
     g_logger.info("OpenGL implementation:\n"
-                  "GL_VENDOR   = %s\n"
-                  "GL_RENDERER = %s\n"
-                  "GL_VERSION  = %s",
+                  "GL_VENDOR    = %s\n"
+                  "GL_RENDERER  = %s\n"
+                  "GL_VERSION   = %s\n"
+                  "GLSL_VERSION = %s\n"
+                  "GLEW_VERSION = %s\n",
                   (const char*)glGetString(GL_VENDOR),
                   (const char*)glGetString(GL_RENDERER),
-                  (const char*)glGetString(GL_VERSION));
-    m_shaderProgram->setContext(context());
-    m_meshFaceShader.reset(new ShaderProgram(context()));
+                  (const char*)glGetString(GL_VERSION),
+                  (const char*)glGetString(GL_SHADING_LANGUAGE_VERSION),
+                  (const char*)glewGetString(GLEW_VERSION));
+
+    // GL_CHECK has to be defined for this to actually do something
+    glCheckError();
+
+    initCursor(10, 1);
+    initAxes();
+    initGrid(2.0f);
+
+    glCheckError();
+
+    m_boundingBoxShader.reset(new ShaderProgram());
+    m_boundingBoxShader->setShaderFromSourceFile("shaders:bounding_box.glsl");
+
+    m_meshFaceShader.reset(new ShaderProgram());
     m_meshFaceShader->setShaderFromSourceFile("shaders:meshface.glsl");
-    m_meshEdgeShader.reset(new ShaderProgram(context()));
+
+    m_meshEdgeShader.reset(new ShaderProgram());
     m_meshEdgeShader->setShaderFromSourceFile("shaders:meshedge.glsl");
-    m_incrementalFramebuffer = allocIncrementalFramebuffer(width(), height());
+
+#ifdef DISPLAZ_USE_QT4
+    double dPR = 1.0;
+#else
+    double dPR = devicePixelRatio();
+#endif
```

Comment:
Yes, done. This situation is not ideal in any case, and another good reason to switch to Qt5 sooner than later.